figures/label_sampling_experiment/plot_scripts/label_sampling_disc_figure_generator.py

```diff
@@ -0,0 +1,187 @@
+                # unstable code
+                # if order of error bars
+                # change then this code will not work
+                for idx, item in enumerate(axes[row_ind][col_ind].get_children()):
+                    # if the points in graph 
+                    # change color map accordingly
+                    if idx == 0 or idx==1:
+                        item.set_edgecolor([
+                            color_map[col] if index==0 else color_map[row] 
+                            for index in range(len(data[col][row][evaluation_set].num_lfs.unique()))
+                        ])
+                        item.set_facecolor([
+                            color_map[col] if index==0 else color_map[row] 
+                            for index in range(len(data[col][row][evaluation_set].num_lfs.unique()))
+                        ])
+
+                    #if error bars change accordingly
+                    elif isinstance(item, plt.Line2D):
+                        if idx == 2:
+                            item.set_linestyle('dashed')
+                            item.set_color("black")
+                            item.set_alpha(0.25)
+                        elif idx == 9:
+                            item.set_linestyle('dashed')
+                            item.set_color("black")
+                            item.set_alpha(0.25)
+                        else:
+                            item.set_color(color_map[row])
+                
+            # only set first column and first row titles
+            if row_ind == 0:
+                axes[row_ind][col_ind].set_title(row, color=color_map[row])
+
+            if col_ind == 0:
+                axes[row_ind][col_ind].set_ylabel(col, color=color_map[col])
+            else:
+                axes[row_ind][col_ind].set_ylabel('')
+
+    for item in axes.flat:
+        item.title.set_fontsize(30)
+        item.yaxis.label.set_fontsize(24)
+        item.xaxis.label.set_fontsize(24)
+        for tick in item.get_yticklabels() + item.get_xticklabels():
+            tick.set_fontsize(23)
+
+    if "label" in data["DaG"]["DaG"]["dev"].columns:
+        axes.flatten()[3].legend(loc='upper center', bbox_to_anchor=(2.53, 0.8), fontsize=20)
+        leg = axes.flatten()[3].get_legend()
+        leg.legendHandles[0].set_edgecolor('black')
+        leg.legendHandles[0].set_facecolor('white')
+        
+        leg.legendHandles[1].set_edgecolor('black')
+        leg.legendHandles[1].set_facecolor('white')
+    
+    fig.text(0.5, 0.89, 'Label Sources', ha='center', fontsize=30)
+    fig.text(0.5, 0.04, 'Number of Additional Label Functions', ha='center', fontsize=30)
+    fig.text(0.04, 0.5, f'Predicted Relations ({metric})', va='center', rotation='vertical', fontsize=25)
+    fig.suptitle(title, fontsize=30)
+    fig.text(0.69, 0.02, '0-Only Uses Relation Specific Databases.', fontsize=27)
+    plt.subplots_adjust(top=0.85)
+    plt.savefig(file_name, format='png')
+
+
+color_map = {
```

Comment:
Not sure if you want to have a map using the english word for these colors for future reference